src/asf_tools/water_map.py

```diff
@@ -312,24 +314,33 @@ def make_water_map(out_raster: Union[str, Path], vv_raster: Union[str, Path], vh
               epsg_code=out_epsg, dtype=gdal.GDT_Byte, nodata_value=nodata)
 
 
-def main():
+def _get_cli(interface: Literal['hyp3', 'main']) -> argparse.ArgumentParser:
     parser = argparse.ArgumentParser(
         description=__doc__,
         formatter_class=argparse.ArgumentDefaultsHelpFormatter
     )
 
-    parser.add_argument('out_raster', help='Water map GeoTIFF to create')
-    # FIXME: Decibel RTCs would be real nice.
-    parser.add_argument('vv_raster',
-                        help='Sentinel-1 RTC GeoTIFF raster, in power scale, with VV polarization')
-    parser.add_argument('vh_raster',
-                        help='Sentinel-1 RTC GeoTIFF raster, in power scale, with VH polarization')
-
-    parser.add_argument('--hand-raster',
-                        help='Height Above Nearest Drainage (HAND) GeoTIFF aligned to the RTC rasters. '
-                             'If not specified, HAND data will be extracted from a Copernicus GLO-30 DEM based HAND.')
-    parser.add_argument('--tile-shape', type=int, nargs=2, default=(100, 100),
-                        help='shape (height, width) in pixels to tile the image to')
+    if interface == 'hyp3':
+        parser.add_argument('--bucket')
+        parser.add_argument('--bucket-prefix', default='')
+        parser.add_argument('--vv-raster',
+                            help='Sentinel-1 RTC GeoTIFF raster, in power scale, with VV polarization.')
+    elif interface == 'main':
+        parser.add_argument('out_raster', help='Water map GeoTIFF to create')
+        # FIXME: Decibel RTCs would be real nice.
+        parser.add_argument('vv_raster',
+                            help='Sentinel-1 RTC GeoTIFF raster, in power scale, with VV polarization')
+        parser.add_argument('vh_raster',
+                            help='Sentinel-1 RTC GeoTIFF raster, in power scale, with VH polarization')
+
+        parser.add_argument('--hand-raster',
+                            help='Height Above Nearest Drainage (HAND) GeoTIFF aligned to the RTC rasters. '
+                                 'If not specified, HAND data will be extracted from the GLO-30 HAND.')
+        parser.add_argument('--tile-shape', type=int, nargs=2, default=(100, 100),
+                            help='shape (height, width) in pixels to tile the image to')
```

Comment:
I don't love this wording either, maybe something like `image tiles will have this shape (height, width) in pixels` would be more clear?
```suggestion
                            help='shape (height, width) in pixels to which to tile the image')
```

Reply:
```suggestion
                            help='image tiles will have this shape (height, width) in pixels')
```
Ooh, I like "image tiles will have this shape (height, width) in pixels" that the bestshape (height, width) in pixels to tile the image to